cmd/ipfs/daemon.go

```diff
@@ -246,6 +250,41 @@ func daemonFunc(req cmds.Request, res cmds.Response) {
 		}
 	}
 
+	// repo blockstore GC
+	const storageMax = 1024 * 1024 * 1024 // 1 GB
+	const storageGcWatermark = storageMax / 10 * 9
+
+	go func() {
+		for {
+			<-time.After(30 * time.Minute)
+
+			// Check fs repo disk usage
+			diskUsage, err := fsrepo.GetDiskUsage()
+			if err != nil {
+				res.SetError(err, cmds.ErrNormal)
+			}
+
+			if diskUsage > storageMax {
+				res.SetError(errors.New("Maximum storage limit exceeded. Maybe unpin some files?"), cmds.ErrNormal)
+			} else if diskUsage > storageGcWatermark {
+				fmt.Println("Starting repo GC...")
+				// 1 minute is sufficient for ~1GB unlink() blocks each of 100kb in SSD
+				// TODO: scale this, e.g. multiply by (diskUsage / 10e9)
+				ctx, cancel := context.WithTimeout(context.TODO(), time.Minute)
```

Comment:
`context.TODO()` here should be derived from the ipfs node's context (or process)